pokemongo_bot/service/player.py

```diff
@@ -0,0 +1,123 @@
+        self._candies = response_dict['candy']
+        self._pokemon = response_dict['pokemon']
+        self._candies = response_dict['candy']
+        self._eggs = response_dict['eggs']
+        self._egg_incubators = response_dict['egg_incubators']
+
+        for item_id in self._inventory:
+            if item_id in self._pokeballs:
+                self._pokeballs[item_id] = self._inventory[item_id]
+
+        return True
+
+    def get_player(self):
+        self.update()
+        return self._player
+
+    def get_inventory(self):
+        self.update()
+        return self._inventory
+
+    def get_eggs(self):
+        self.update()
+        return self._eggs
+
+    def get_egg_incubators(self):
+        self.update()
+        return self._egg_incubators
+
+    def get_pokemon(self):
+        self.update()
+        return self._pokemon
+
+    def get_candies(self):
+        self.update()
+        return self._candies
+
+    def get_candy(self, pokemon_id):
+        self.update()
+        try:
+            return self._candies[pokemon_id]
+        except KeyError:
+            return 0
+
+    def add_candy(self, pokemon_id, pokemon_candies):
+        pokemon_id = int(pokemon_id)
+        if pokemon_id in self._candies:
+            self._candies[pokemon_id] += int(pokemon_candies)
+        else:
+            self._candies[pokemon_id] = int(pokemon_candies)
+
+    def get_pokeballs(self):
+        self.update()
+        return self._pokeballs
+
+    def print_stats(self):
+        if self.update() is True:
+            logger.log('', prefix='#')
+            logger.log('Username: {}'.format(self._player.username), prefix='#')
+            logger.log('Account creation: {}'.format(self._player.get_creation_date()), prefix='#')
+            logger.log('Bag storage: {}/{}'.format(self._inventory['count'], self._player.max_item_storage), prefix='#')
+            logger.log('Pokemon storage: {}/{}'.format(len(self._pokemon) + len(self._eggs), self._player.max_pokemon_storage), prefix='#')
+            logger.log('Stardust: {:,}'.format(self._player.stardust), prefix='#')
+            logger.log('Pokecoins: {}'.format(self._player.pokecoin), prefix='#')
+            logger.log('Poke Balls: {}'.format(self._pokeballs[1]), prefix='#')
+            logger.log('Great Balls: {}'.format(self._pokeballs[2]), prefix='#')
+            logger.log('Ultra Balls: {}'.format(self._pokeballs[3]), prefix='#')
+            logger.log('-- Level: {}'.format(self._player.level), prefix='#')
+            logger.log('-- Experience: {:,}'.format(self._player.experience), prefix='#')
+            logger.log(
+                '-- Experience until next level: {:,}'.format(self._player.next_level_xp - self._player.experience), prefix='#')
+            logger.log('-- Pokemon captured: {:,}'.format(self._player.pokemons_captured), prefix='#')
+            logger.log('-- Pokestops visited: {:,}'.format(self._player.poke_stop_visits), prefix='#')
```

Comment:
Instead of adding `prefix='#'` to all of these, define a new function here instead:

```
def stats_log(output):
    logger.log(output, prefix='#')
```

Easier to change prefix later